test/extensions/common/wasm/wasm_test.cc

```diff
@@ -1399,6 +1399,63 @@ TEST_P(WasmCommonContextTest, EmptyContext) {
   root_context_->validateConfiguration("", plugin_);
 }
 
+class TestFilter : public Envoy::Extensions::Common::Wasm::Context {
+public:
+  TestFilter(Wasm* wasm, uint32_t root_context_id, PluginHandleSharedPtr plugin_handle)
+      : Envoy::Extensions::Common::Wasm::Context(wasm, root_context_id, plugin_handle) {}
+  using ::Envoy::Extensions::Common::Wasm::Context::log;
+  proxy_wasm::WasmResult log(uint32_t, std::string_view message) override {
+    std::cerr << message << "\n";
+    return proxy_wasm::WasmResult::Ok;
+  }
+};
+
+class WasmCommonHttpContextTest
+    : public Common::Wasm::WasmHttpFilterTestBase<
+          testing::TestWithParam<std::tuple<std::string, std::string>>> {
+public:
+  WasmCommonHttpContextTest() = default;
+
+  void setup(const std::string& code, std::string vm_configuration, std::string root_id = "") {
+    setRootId(root_id);
+    setVmConfiguration(vm_configuration);
+    setupBase(std::get<0>(GetParam()), code,
+              [](Wasm* wasm, const std::shared_ptr<Plugin>& plugin) -> ContextBase* {
+                return new NiceMock<TestContext>(wasm, plugin);
+              });
+  }
+  void setupFilter() { setupFilterBase<TestFilter>(); }
+
+  TestContext& rootContext() { return *static_cast<TestContext*>(root_context_); }
+
+  TestFilter& filter() { return *static_cast<TestFilter*>(context_.get()); }
+};
+
+INSTANTIATE_TEST_SUITE_P(Runtimes, WasmCommonHttpContextTest,
+                         Envoy::Extensions::Common::Wasm::runtime_and_cpp_values);
+
+TEST_P(WasmCommonHttpContextTest, DuplicateLocalReply) {
+  std::string code;
+  if (std::get<0>(GetParam()) != "null") {
+    code = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(absl::StrCat(
+        "{{ test_rundir }}/test/extensions/common/wasm/test_data/test_context_cpp.wasm")));
+  } else {
+    // The name of the Null VM plugin.
+    code = "CommonWasmTestContextCpp";
+  }
+  EXPECT_FALSE(code.empty());
+
+  setup(code, "duplicate_local_reply");
```

Comment:
This kind of conditioning of setup on test name is a bit of an anti-pattern. I would prefer whatever setup criteria parameters there are to reflect the semantic intent, rather than the caller test's name.

Reply:
Thanks for your advice